#!/usr/bin/env python
# Copyright (c) 2016 The UUV Simulator Authors.
# All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

PACKAGE = "uuv_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("linear_p", double_t, 0, "p component of pid for linear vel.", 1., 0, 100)
gen.add("linear_i", double_t, 0, "i component of pid for linear vel.", 0.0, 0, 100)
gen.add("linear_d", double_t, 0, "d component of pid for linear vel.", 0.0, 0, 100)
gen.add("linear_sat", double_t, 1, "saturation of pid for linear vel.", 10.0, 0, 100)

gen.add("angular_p", double_t, 0, "p component of pid for angular vel.", 1., 0, 100)
gen.add("angular_i", double_t, 0, "i component of pid for angular vel.", 0.0, 0, 100)
gen.add("angular_d", double_t, 0, "d component of pid for angular vel.", 0.0, 0, 100)
gen.add("angular_sat", double_t, 1, "saturation of pid for angular vel.", 3.0, 0, 100)

gen.add("odom_vel_in_world", bool_t, 0, "Is odometry velocity supplied in world frame? (gazebo)", True)

exit(gen.generate(PACKAGE, "uuv_control_cascaded_pid", "VelocityControl"))
